### PR TITLE
PR submission checklist: Add reminder to check self-hosted sites

### PR DESCRIPTION
**Pull request by momo-ozawa**
Addresses pcdRpT-5mW-p2#comment-8790

## Description
Adds a reminder to test a feature for both WP.com and self-hosted sites

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ PR submission checklist:
 - [ ] I have completed the Regression Notes.
 - [ ] I have considered adding unit tests for my changes.
 - [ ] I have considered adding accessibility improvements for my changes.
+- [ ] I have tested this feature for both WordPress.com sites and self-hosted Jetpack sites.
```
Comment (kean) [inline .github/PULL_REQUEST_TEMPLATE.md]: It seems a bit too specific for the PR submission checklist.
It sounds like part of the testing instructions. Would you consider adding it under the "Testing checklist" (renamed from "UI Changes testing checklist" below?

Comment (momo-ozawa) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done! [fa386ec](https://github.com/wordpress-mobile/WordPress-iOS/pull/22493/commits/fa386ec9ed6dff8907f1e3bd6c385f106c108df2)

**Review by kean — APPROVED** [review]
It may be worth asking a larger group for a review.
To me, it seems like a clearly good thing to add.

**Comment by momo-ozawa** [comment]
@kean Thanks! I've published an RFC here pcdRpT-5Eb-p2

**Pull request merged**